pkg/render/tpl.go

```diff
@@ -0,0 +1,83 @@
+package render
+
+import (
+	"sync"
+	"text/template"
+
+	"github.com/golang/groupcache/lru"
+)
+
+//go:generate mockgen -source=tpl.go -destination=tpl_mock.go -package=render   TplProvider
+type TplProvider interface {
```

Comment:
- this interface is not good, it is depended on `template.Template`
- it looks like we could merge `ParseTplProvider ` and `CachedTplProvider ` to  `TplRender, not we split it but does not bring any benefit, make that sure abstraction is meaningful

Reply:
merged

Reply:
nice, so now we can remove the `TplProvider` interface.